sharedbuffers/mapped_struct.py

```diff
@@ -1976,8 +1975,27 @@ def _intersect_2(self, seq):
         elif not seq:
             return proxied_frozenset(None, 0, 0)
 
-        rv = self._add_to(set())
-        rv.intersection_update(seq)
+        if len(self) > len(seq):
+            rv = set(seq)
+            typ = type(seq)
```

Comment:
Don't assign it to a local, that hurts rather than help. Cython knows how to optimize `type(x) is y` very well.